MoneySplitter.Services/Api/QueryApiService.cs

```diff
@@ -57,11 +51,25 @@ public class QueryApiService : IQueryApiService
             TResultQuery resultQuery;
             using (var httpClient = new HttpClient())
             {
-                var responce = await httpClient.GetStringAsync(uri);
-                resultQuery = JsonConvert.DeserializeObject<TResultQuery>(responce);
+                var responce = await httpClient.GetAsync(uri);
+                resultQuery = await GetContentResponce<TResultQuery>(responce);
             }
 
             return resultQuery;
         }
+
+        private async Task<TContentResponce> GetContentResponce<TContentResponce>(HttpResponseMessage responseMessage)
+            where TContentResponce : class
+        {
+            if (responseMessage.StatusCode != HttpStatusCode.OK)
+            {
+                throw new Exception();
```

Comment:
Don't throw not informative exceptions. Use special types of exceptions or message

Reply:
add info